src/basics.md

```diff
@@ -270,6 +271,54 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-regex-filter-log]: #ex-regex-filter-log
+<a name="ex-regex-filter-log"></a>
+## Filter a log file by matching multiple regular expressions
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Reads a file named `application.log` and only outputs the lines 
+containing “warning” or “error” (case insensitive). A [`regex::RegexSet`] is built
+with [`regex::RegexSetBuilder`] using the builder pattern.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+# use std::fs::File;
+# use std::io::{BufReader, BufRead};
+# 
+use regex::RegexSetBuilder;
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Regex(regex::Error);
+#     }
+# }
+#
+fn run() -> Result<()> {
+    let log_path = "application.log";
+    let log_file = File::open(log_path)?;
+    let buffered = BufReader::new(log_file);
+
+    let set = RegexSetBuilder::new(&[r"warning", r"error"])
```

Comment:
how about adding more complicated regexes here to make the example more real life. Just think of the last loggs you have greped and construct something similar ;)